### PR TITLE
Store

### PR DESCRIPTION
**Pull request by awaisanwar544**
1. Add funcionality for "Add new" button click
2. add a funcionality for "Remove" button click
3. add a funcionality for "Check status" button click

https://user-images.githubusercontent.com/87834222/160711709-19510a5c-776f-420b-b0a7-aae7fd6177a6.mov

### CODE REVIEW

File: src/components/AddBookForm.jsx [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+    dispatch(addBook({
+      id: Date.now(),
+      title: event.target[0].value,
+      author: event.target[1].value,
+    }));
+    event.target.reset();
+  };
+
+  return (
+    <form onSubmit={(event) => { handleSubmit(event); }}>
```
Comment (GzimAsani) [inline src/components/AddBookForm.jsx]: - [x] You did a good job here, but in the project requirements the submission should be done through a button, so please create a button that creates new books. You can accomplish this by using a local state to track changes.

File: src/components/AddBookForm.jsx [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
-    <input type="submit" value="Add Book" />
-  </form>
-);
+const AddBookForm = () => {
+  const dispatch = useDispatch();
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    dispatch(addBook({
+      id: Date.now(),
```
Comment (GzimAsani) [inline src/components/AddBookForm.jsx]: - [x] This id should be unique

File: src/components/BookCard.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/books';
+
+const BookCard = ({ title, author, id }) => {
+  const dispatch = useDispatch();
+  return (
+    <div className="book">
+      <h3>{title}</h3>
+      <p>{author}</p>
+      <input
```
Comment (GzimAsani) [inline src/components/BookCard.jsx]: - [x] The remove functionality should also be done through a `button` as it was specified in the project requirement

File: src/components/Categories.jsx [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
-const Categories = () => (
-  <div className="categories" />
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  const status = useSelector((state) => state.status);
+  return (
+    <div className="categories">
+      <p>{status}</p>
+      <input
```
Comment (GzimAsani) [inline src/components/Categories.jsx]: - [x] Again, this is working fine, but it is a project requirement to do this through a button. So please update this

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi  @awaisanwar544          
Good job so far! You are almost done :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the git-flow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [x] _Check the comments below :arrow_double_down: _

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
Hi @awaisanwar544,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**